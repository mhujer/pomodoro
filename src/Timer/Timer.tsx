import React from 'react';
import { formatPomodoroTime } from '../timeFormatUtils';

type Props = {
    remainingSeconds: number;
}

const Timer: React.FC<Props> = ({ remainingSeconds }) => {
    const formattedSeconds = formatPomodoroTime(remainingSeconds);

    return (
        <>
            <p className='countdown'>{formattedSeconds}</p>
        </>
    );

};

export default Timer;
