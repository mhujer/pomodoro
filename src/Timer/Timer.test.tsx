import React from 'react';
import Timer from './Timer';
import renderer from 'react-test-renderer';

describe('Timer', () => {
    test('renders timer for 0 seconds', () => {
        const domTree = renderer.create(<Timer remainingSeconds={0} />).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    test('renders timer for 60 seconds', () => {
        const domTree = renderer.create(<Timer remainingSeconds={60} />).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    test('renders timer for 1500 seconds (25 min)', () => {
        const domTree = renderer.create(<Timer remainingSeconds={1500} />).toJSON();
        expect(domTree).toMatchSnapshot();
    });
});
