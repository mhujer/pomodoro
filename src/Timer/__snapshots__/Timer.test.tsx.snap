// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders timer for 0 seconds 1`] = `
<p
  className="countdown"
>
  00:00
</p>
`;

exports[`renders timer for 60 seconds 1`] = `
<p
  className="countdown"
>
  01:00
</p>
`;

exports[`renders timer for 1500 seconds (25 min) 1`] = `
<p
  className="countdown"
>
  25:00
</p>
`;
